ns1/resource_record.go

```diff
@@ -338,6 +338,22 @@ func resourceDataToRecord(r *dns.Record, d *schema.ResourceData) error {
 
 			if v, ok := answer["meta"]; ok {
 				log.Println("answer meta", v)
+				if allsubdivisions, ok := v.(map[string]interface{})["subdivisions"]; ok {
+					subdivisions := strings.Split(allsubdivisions.(string), ",")
+					subdvisionsMap := make(map[string]interface{})
+					for _, sub := range subdivisions {
+						subp := strings.Split(sub, "-")
+						if len(subp) != 2 {
+							return fmt.Errorf("invalid subidivision format. expecting (\"Country-Subdivision\") got %s", sub)
+						}
+						subdvisionsMap[subp[0]] = []string{}
+					}
+					for _, sub := range subdivisions {
+						sub := strings.Split(sub, "-")
+						subdvisionsMap[sub[0]] = append(subdvisionsMap[sub[0]].([]string), sub[1])
```

Comment:
I might be missing something, but I don't think I understand why we need to perform a 2nd iteration through this list and split again to perform this step.  Why not just move this to the end of the previous loop?

```
	for _, sub := range subdivisions {
		subp := strings.Split(sub, "-")
		if len(subp) != 2 {
			return fmt.Errorf("invalid subidivision format. expecting (\"Country-Subdivision\") got %s", sub)
		}
		if subdvisionsMap[subp[0]] == nil {
			subdvisionsMap[subp[0]] = []string{}
		}
		subdvisionsMap[subp[0]] = append(subdvisionsMap[subp[0]].([]string), subp[1])
	}
```

Reply:
@mburtless you're right, thanks